[skip-ci] mention auto deprecation in release notes (#17836)
root-project/root: README/ReleaseNotes/v632/index.md

- Support for external (ie. non-builtin) libAfterImage is now deprecated and it will be removed in next release 6.34.
- The `TList::TList(TObject*)` constructor is deprecated and will be removed in ROOT 6.34
- The deprecated `TProofOutputList::TProofOutputList(TObject *o)` constructor was removed

> - In the ROOT terminal, CINT-like declarations `i=0` are now deprecated and might be no longer supported in future ROOT releases. Please use instead `auto i=0`. In the meantime, you can still use it and this warning will be emitted: `declaration without the 'auto' keyword is deprecated`.

Reviewer: ```suggestion
- In the ROOT terminal, CINT-like declarations such as `i=0` are now deprecated and might be no longer supported in future ROOT releases. Please use instead the correct C++ syntax `auto i=0`. In the meantime, you can still use it and this warning will be emitted: `declaration without the 'auto' keyword is deprecated`.
```

> - In the ROOT terminal, CINT-like declarations such as `i=0` are now deprecated and might be no longer supported in future ROOT releases. Please use instead the correct C++ syntax `auto i=0`. In the meantime, you can still use it and this warning will be emitted: `declaration without the 'auto' keyword is deprecated`.

Author: ```suggestion
- In the ROOT terminal or in interpreted scripts, CINT-like declarations such as `i=0` are now deprecated and might be no longer supported in future ROOT releases. Please use instead the correct C++ syntax `auto i=0`. In the meantime, you can still use it and this warning will be emitted: `declaration without the 'auto' keyword is deprecated`.
```

> - In the ROOT terminal or in interpreted scripts, CINT-like declarations such as `i=0` are now deprecated and might be no longer supported in future ROOT releases. Please use instead the correct C++ syntax `auto i=0`. In the meantime, you can still use it and this warning will be emitted: `declaration without the 'auto' keyword is deprecated`.

Reviewer 2: ```suggestion
- On the ROOT prompt or in interpreted scripts, CINT-like implicit declarations such as `i = 0;` are now deprecated and might no longer be supported in future ROOT releases.   Update those declarations by explicitly using the `auto` keyword: `auto i=0`.    The deprecation is marked by printing this kind of warning: `declaration without the 'auto' keyword is deprecated`.
```
